### ISSUE TITLE
AGALMiniAssembler doesn't support array vc lookup

### ISSUE THREAD

**Issue by peteshand** [openfl/openfl#1262]
The following is valid AGAL
vc[va1.x].x 

This works in the adobe version of AGALMiniAssembler and is a really useful feature for things like highly dynamic particle systems. Any idea who ported this code and why they decided to leave it out?

It looks like it would be done in the emitSource function, however doesn't look to be complete.

**Comment by peteshand** [comment]
Looks like it's an issue with this regex
var indexed = getGroupMatches (~/vc\[(v[tcai])(\d+)\.([xyzw])([\+\-]\d+)?\](\.[xyzw]{1,4})?/i, token, 5);

If I pass in:
vc[vt1.w].xyz
I would expect to get an array of
["vc[vt1.w].xyz", "vt", "1", "w", "vc", ".xyz"]
however the last two strings are not present
["vc[vt1.w].xyz", "vt", "1", "w"]

Anyone know their regular expressions?

**Comment by ibilon** [comment]
The expression isn't correct, the first `[` should be escaped, could that be the issue?

**Comment by jgranick** [comment]
Hi Pete,

I started digging into your issue, and found that our `AGALMiniAssembler` did not resemble the Adobe version very much. This was probably due to the old Stage3D implementation, which did not rely upon AGAL byte code, but source code.

Since we have a byte code-based new implementation, there's no reason for `AGALMiniAssembler` to be different from the original version. I just ported it from ActionScript, and it appears to be working fine in all of my tests

https://github.com/openfl/openfl/commit/1e37e00dde8589b82492fc5a1cbb7710a3c2c4fd

I'm going to close this (because I assume this is all resolved) but let me know if there are any regressions with the new assembler port :smile:
